File: src/models/SavedLink.js
import mongoose from 'mongoose';

const SavedLinkSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
  url: { type: String, required: true, trim: true },
  title: { type: String, required: true, trim: true, maxlength: 200 },
  description: { type: String, default: '', maxlength: 500 },
  category: {
    type: String,
    enum: ['tutorial', 'article', 'documentation', 'tool', 'job-posting', 'video', 'course', 'github', 'other'],
    default: 'other',
  },
  tags: [{ type: String, trim: true, lowercase: true }],
  favicon: { type: String, default: '' },
  isPinned: { type: Boolean, default: false },
  isArchived: { type: Boolean, default: false },
}, { timestamps: true });

SavedLinkSchema.index({ userId: 1, category: 1, createdAt: -1 });

export default mongoose.models.SavedLink || mongoose.model('SavedLink', SavedLinkSchema);

File: backend-spring/src/main/java/com/nextlevel/api/model/SavedLink.java
package com.nextlevel.api.model;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "savedlinks")
@CompoundIndex(def = "{'userId': 1, 'category': 1, 'createdAt': -1}")
public class SavedLink {

    @Id
    private String id;
    private String userId;
    private String url;
    private String title;
    private String description = "";
    private String category = "other";
    private List<String> tags = new ArrayList<>();
    private String favicon = "";
    private Boolean isPinned = false;
    private Boolean isArchived = false;
    private Instant createdAt;
    private Instant updatedAt;
}
